### DISCUSSION TITLE
Relationship between OIDCSessionInactivityTimeout and Keycloak SSO Session Idle timeout

### DISCUSSION THREAD

**Discussion by blevine** [OpenIDC/mod_auth_openidc#909] [Q&A] [answered]
Could someone explain the relationship between these two settings? Does one take precedence over the other? What is the session expiry behavior if they are not set to the same value? In general, I would like the Keycloak setting to govern when the client's session is invalidated. Thanks!

**Comment by zandbelt** [comment]
they are not related, they are independent of each other; one tracks the idle time on the Apache/mod_auth_openidc application session, the other tracks the idle time on the Keycloak SSO session; you can set the value to the same timeout, but one does not (and can not) actually govern the other

**Reply by blevine** [reply]
I'm trying to understand the authentication behavior in these scenarios.

Assume OIDCSessionInactivityTimeout = 5 minutes and Keycloak Session Idle Timeout = 10 minutes. If the there is no user interaction for 6 minutes, I assume that the user is not redirected to the login page (since the SSO session has not expired).  Is that correct?  At that point is a new mod_auth_open_idc session created?

Conversely, assume OIDCSessionInactivityTimeout = 10 minutes and Keycloak Session Idle Timeout = 5 minutes. If there is no user interaction for 6 minutes, I assume that the user is not redirected to the login page because there is still a valid mod_auth_openidc session. Correct? But at this point, the SSO session has expired?

**Reply by zandbelt** [reply]
that is all correct indeed

**Comment by zandbelt** [comment, marked as answer]
see also: https://github.com/zmartzone/mod_auth_openidc/wiki/Sessions-and-Timeouts

**Reply by blevine** [reply]
Very helpful.  Thank you!
